### PR TITLE
Fix CommonCrypto headers to compile as Objective C

### PR DESCRIPTION
**Pull request by rajsesh-msft**
bool isn't defined for ObjC. Fix that and add unit tests (just compile) headers as OBJC (our default is OBJCPP, hence we missed this).

### CODE REVIEW

**Comment by rajsesh-msft** [comment]
@bbowman is added to the review. #Closed

**Comment by rajsesh-msft** [comment]
@jofre-ms is added to the review. #Closed

File: tests/unittests/Starboard/CommonCryptoTests.m [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//******************************************************************************
+//
+// Copyright (c) 2015 Microsoft Corporation. All rights reserved.
```
Comment (msft-Jeyaram) [inline tests/unittests/Starboard/CommonCryptoTests.m]: super nit: 2016

File: build/Tests/UnitTests/Starboard/Starboard.UnitTests.vcxproj [hunk 2]
```diff
@@ -1,247 +1,250 @@
...
+    </ClangCompile>
+  </ItemGroup>
+  <ItemGroup>
+    <ClInclude Include="..\..\..\..\tests\unittests\Starboard\LifetimeCounting.h" />
+  </ItemGroup>
+  <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+  <ImportGroup Label="ExtensionTargets">
+    <Import Project="$(StarboardBasePath)\msvc\starboard-cmdline.targets" />
+  </ImportGroup>
+</Project>
```
Comment (DHowett-MSFT) [inline build/Tests/UnitTests/Starboard/Starboard.UnitTests.vcxproj]: github is showing this file as 100% changed; has its encoding or line ending style changed?

File: include/CommonCrypto/CommonCryptor.h [hunk 3]
```diff
@@ -28,6 +28,10 @@
 #define CC_SHA384_DIGEST_LENGTH 48
 #define CC_SHA512_DIGEST_LENGTH 64
 
+#ifndef __cplusplus
+typedef int8_t bool;
```
Comment (DHowett-MSFT) [inline include/CommonCrypto/CommonCryptor.h]: Does it make more sense for us to include stdbool.h here? It does the same (similar: `bool -> _Bool` from C99) typedef in a C++ guard.

**Comment by DHowett-MSFT** [comment]
[nit] It looks like GH wants to pull in a lot of the merged changes; can you rebase this to clean up the diff?

File: build/Tests/UnitTests/Starboard/Starboard.UnitTests.vcxproj (hunk 2 above)
Comment (rajsesh-msft) [inline build/Tests/UnitTests/Starboard/Starboard.UnitTests.vcxproj]: That is strange, the encoding hasn't changed.  The only change here is addition of this file.
  <ClangCompile Include="..\..\..\..\tests\unittests\Starboard\CommonCryptoTests.m">
      <CompileAs>CompileAsObjc</CompileAs>
    </ClangCompile>

---
In reply to: [70302186](https://github.com/Microsoft/WinObjC/pull/629#discussion_r70302186) [](ancestors = 70302186)

Comment (DHowett-MSFT) [inline build/Tests/UnitTests/Starboard/Starboard.UnitTests.vcxproj]: Ah, it looks like the line endings changed UNIX to Windows.

---
In reply to: [70302644](https://github.com/Microsoft/WinObjC/pull/629#discussion_r70302644) [](ancestors = 70302644,70302186)

**Comment by jofre-ms** [comment]
:shipit:

File: build/Tests/UnitTests/Starboard/Starboard.UnitTests.vcxproj (hunk 2 above)
Comment (rajsesh-msft) [inline build/Tests/UnitTests/Starboard/Starboard.UnitTests.vcxproj]: Oh, the joys of editing in VS.

---
In reply to: [70303167](https://github.com/Microsoft/WinObjC/pull/629#discussion_r70303167) [](ancestors = 70303167,70302644,70302186)

**Comment by rajsesh-msft** [comment]
@DHowett-MSFT @msft-Jeyaram the git surgery lost the other comments.  The issues are all fixed now.

---
In reply to: [231808108](https://github.com/Microsoft/WinObjC/pull/629#issuecomment-231808108) [](ancestors = 231808108)

**Comment by msft-Jeyaram** [comment]
:shipit:

**Comment by DHowett-MSFT** [comment]
:shipit:

**Pull request merged**